private/buf/bufprint/repository_label_printer.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufprint
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	modulev1 "buf.build/gen/go/bufbuild/registry/protocolbuffers/go/buf/registry/module/v1"
+)
+
+type repositoryLabelPrinter struct {
+	writer io.Writer
+}
+
+func newRepositoryLabelPrinter(
+	writer io.Writer,
+) *repositoryLabelPrinter {
+	return &repositoryLabelPrinter{
+		writer: writer,
+	}
+}
+
+func (p *repositoryLabelPrinter) PrintRepositoryLabel(ctx context.Context, format Format, message *modulev1.Label) error {
+	outRepositoryLabel := registryLabelToOutputLabel(message)
+	switch format {
+	case FormatText:
+		return p.printRepositoryLabelsText([]outputRepositoryLabel{outRepositoryLabel})
+	case FormatJSON:
+		return json.NewEncoder(p.writer).Encode(outRepositoryLabel)
+	default:
+		return fmt.Errorf("unknown format: %v", format)
+	}
+}
+
+func (p *repositoryLabelPrinter) PrintRepositoryLabels(ctx context.Context, format Format, nextPageToken string, messages ...*modulev1.Label) error {
+	if len(messages) == 0 {
+		return nil
+	}
+	outputRepositoryLabels := registryLabelsToOutputLabels(messages)
+	switch format {
+	case FormatText:
+		return p.printRepositoryLabelsText(outputRepositoryLabels)
+	case FormatJSON:
+		return json.NewEncoder(p.writer).Encode(paginationWrapper{
+			NextPage: nextPageToken,
+			Results:  outputRepositoryLabels,
+		})
+	default:
+		return fmt.Errorf("unknown format: %v", format)
+	}
+}
+
+func (p *repositoryLabelPrinter) printRepositoryLabelsText(outputRepositoryLabels []outputRepositoryLabel) error {
+	return WithTabWriter(
+		p.writer,
+		[]string{
+			"Name",
+			"Commit",
+			"Create Time",
+			"Archive Time",
+		},
+		func(tabWriter TabWriter) error {
+			for _, outputRepositoryLabel := range outputRepositoryLabels {
+				formattedArchiveTime := "N/A"
```

Comment:
Unarchived (normal) labels don't have an archived time, so I'm just printing a placeholder here. I could also just print an empty string here, but it could look weird if all labels printed are unarchived.

Another solution is to print this column only when at least one commit has an archived time.

Reply:
Would prefer empty string. Even better, would prefer column not be printed in text mode if there are no labels with an archive time.